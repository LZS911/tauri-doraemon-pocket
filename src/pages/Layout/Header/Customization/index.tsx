import React from 'react';
import {
  ThemeModeEnum,
  ColorSchemeEnum,
  FontFamilyEnum,
} from '../../../../common/enum';
import i18n from '../../../../locale';
import Customization from './Customization';
export interface CollapseContentProps<T> {
  value?: T;
  defaultValue?: T;
  onChange?: (val?: T) => void;
  list: Array<{
    imgSrc?: string;
    content?: React.ReactNode;
    describe?: string;
    value: T;
    key: string;
    imgClass?: string;
  }>;
}
export enum CollapseKeyEnum {
  THEME_LAYOUT,
  THEME_MODE,
  COLOR_SCHEME,
  FONT_FAMILY,
}
export enum ThemeLayoutTypeEnum {
  DEFAULT,
  MINI,
}
export const themeLayoutTypeList: () => CollapseContentProps<ThemeLayoutTypeEnum>['list'] =
  () => [
    {
      key: 'default',
      describe: i18n.t(
        'layout.header.customization.themeLayoutPanel.defaultType'
      ),
      value: ThemeLayoutTypeEnum.DEFAULT,
      content: <img src="/static/images/layout-default.svg" />,
    },
    {
      key: 'mini',
      describe: i18n.t('layout.header.customization.themeLayoutPanel.miniType'),
      value: ThemeLayoutTypeEnum.MINI,
      content: <img src="/static/images/layout-mini.svg" />,
    },
  ];
export const themeModeList: () => CollapseContentProps<ThemeModeEnum>['list'] =
  () => [
    {
      key: 'light',
      describe: i18n.t('layout.header.customization.themeModePanel.light'),
      value: ThemeModeEnum.Light,
      content: <img src="/static/images/theme-light.svg" />,
    },
    {
      key: 'dark',
      describe: i18n.t('layout.header.customization.themeModePanel.dark'),
      value: ThemeModeEnum.Dark,
      imgSrc: '/static/images/theme-dark.svg',
      content: <img src="/static/images/theme-dark.svg" />,
    },
  ];
export const colorSchemeList: () => CollapseContentProps<ColorSchemeEnum>['list'] =
  () => {
    return [
      {
        key: 'blue',
        describe: i18n.t(
          'layout.header.customization.colorSchemePanel.default'
        ),
        value: ColorSchemeEnum.Blue,
        content: (
          <img
            src="/static/images/color-scheme-blue.svg"
            className="bg-[#1890ff] dark:bg-[#177ddc]"
          />
        ),
      },
      {
        key: 'green',
        describe: i18n.t('layout.header.customization.colorSchemePanel.theme1'),
        value: ColorSchemeEnum.Green,
        content: (
          <img
            src="/static/images/color-scheme-green.svg"
            className="bg-[#00a854] dark:bg-[#03934c]"
          />
        ),
      },
      {
        key: 'purple',
        describe: i18n.t('layout.header.customization.colorSchemePanel.theme2'),
        value: ColorSchemeEnum.Purple,
        content: (
          <img
            src="/static/images/color-scheme-purple.svg"
            className="bg-[#7265e6] dark:bg-[#6559c7]"
          />
        ),
      },
    ];
  };
export const fontFamilyList: () => CollapseContentProps<FontFamilyEnum>['list'] =
  () => [
    {
      key: 'default',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.default'),
      value: FontFamilyEnum.Default,
      content: (
        <>
          <div className="w-14 font-sans font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'inter',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.inter'),
      value: FontFamilyEnum.Inter,
      content: (
        <>
          <div className="w-14 font-inter font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'poppins',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.poppins'),
      value: FontFamilyEnum.Poppins,
      content: (
        <>
          <div className="w-14 font-poppins font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'roboto',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.roboto'),
      value: FontFamilyEnum.Roboto,
      content: (
        <>
          <div className="w-14 font-roboto font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'serif',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.serif'),
      value: FontFamilyEnum.Serif,
      content: (
        <>
          <div className="w-14 font-serif font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'mono',
      describe: i18n.t('layout.header.customization.fontFamilyPanel.mono'),
      value: FontFamilyEnum.Mono,
      content: (
        <>
          <div className="w-14 font-mono font-bold">Aa</div>
        </>
      ),
    },
    {
      key: 'publicSans',
      describe: i18n.t(
        'layout.header.customization.fontFamilyPanel.publicSans'
      ),
      value: FontFamilyEnum.PublicSans,
      content: (
        <>
          <div className="font-publicSans font-bold">Aa</div>
        </>
      ),
    },
  ];
export default Customization;
