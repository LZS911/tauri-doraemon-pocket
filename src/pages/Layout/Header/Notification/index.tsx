import {
  CloseCircleOutlined,
  ExclamationCircleOutlined,
  SettingOutlined,
  WarningOutlined,
} from '@ant-design/icons';
import i18n from '../../../../locale';
import Notification from './Notification';

export default Notification;

export enum NotificationTypeEnum {
  system = 'system',
  information = 'information',
  error = 'error',
  warn = 'warn',
}
export enum NotificationStatusEnum {
  read = 'read',
  unread = 'unread',
}
export interface INotificationList {
  id: number;
  status: NotificationStatusEnum;
  message: string;
  desc: string;
  time: string;
  type: NotificationTypeEnum;
}

export const mockNotificationList = () => {
  return Promise.resolve<INotificationList[]>([
    {
      id: 1,
      status: NotificationStatusEnum.unread,
      message: i18n.t('backend.notification.message1'),
      desc: i18n.t('backend.notification.desc1'),
      time: '8:00 AM',
      type: NotificationTypeEnum.system,
    },
    {
      id: 2,
      status: NotificationStatusEnum.unread,
      message: i18n.t('backend.notification.message2'),
      desc: i18n.t('backend.notification.desc2'),
      time: '10:00 AM',
      type: NotificationTypeEnum.information,
    },
    {
      id: 3,
      status: NotificationStatusEnum.read,
      message: i18n.t('backend.notification.message3'),
      desc: i18n.t('backend.notification.desc3'),
      time: '1:00 PM',
      type: NotificationTypeEnum.error,
    },
    {
      id: 4,
      status: NotificationStatusEnum.read,
      message: i18n.t('backend.notification.message4'),
      desc: i18n.t('backend.notification.desc4'),
      time: '4:00 PM',
      type: NotificationTypeEnum.warn,
    },
  ]);
};

export const genTypeIcon = (type: NotificationTypeEnum) => {
  const typeCorrespondIconMap = new Map<NotificationTypeEnum, JSX.Element>([
    [
      NotificationTypeEnum.system,
      <div
        className="flex h-7 w-7 items-center justify-center rounded-[50%] bg-violet-50"
        key="system"
      >
        <SettingOutlined className="text-violet-500" />
      </div>,
    ],
    [
      NotificationTypeEnum.error,
      <div
        className="flex h-7 w-7 items-center justify-center rounded-[50%] bg-red-50"
        key="system"
      >
        <CloseCircleOutlined className="text-red-500" />
      </div>,
    ],
    [
      NotificationTypeEnum.information,
      <div
        className="flex h-7 w-7 items-center justify-center rounded-[50%] bg-slate-50"
        key="system"
      >
        <ExclamationCircleOutlined className="text-slate-500" />
      </div>,
    ],
    [
      NotificationTypeEnum.warn,
      <div
        className="flex h-7 w-7 items-center justify-center rounded-[50%] bg-yellow-50"
        key="system"
      >
        <WarningOutlined className="text-yellow-500" />
      </div>,
    ],
  ]);

  return typeCorrespondIconMap.get(type);
};
