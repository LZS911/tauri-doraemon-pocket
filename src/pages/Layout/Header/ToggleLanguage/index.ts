import i18n, { SupportLanguage } from '../../../../locale';
import ToggleLanguage from './ToggleLanguage';

export default ToggleLanguage;

export const genLanguageOptions = () => {
  return [
    {
      key: 'zhCN',
      value: SupportLanguage.zhCN,
      text: i18n.t('common.chinese'),
    },
    {
      key: 'enUS',
      value: SupportLanguage.enUS,
      text: i18n.t('common.english'),
    },
  ];
};
