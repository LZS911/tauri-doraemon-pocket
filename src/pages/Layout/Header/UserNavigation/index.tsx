import {
  EditOutlined,
  FormOutlined,
  LockOutlined,
  MailOutlined,
  UserOutlined,
} from '@ant-design/icons';
import i18n from '../../../../locale';
import UserNavigation from './UserNavigation';
export interface ITabWrapperDataSource {
  icon: JSX.Element;
  onClick: () => void;
  text: string;
  hidden?: boolean;
}
export interface UserNavigationTabWrapperProps {
  dataSource: ITabWrapperDataSource[];
}
export enum UserNavigationTabsKeyEnum {
  profile,
  settings,
}
export const UserProfileTabDataSource: () => ITabWrapperDataSource[] = () => [
  {
    icon: <EditOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.profile.editProfile'),
  },
  {
    icon: <UserOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.profile.viewProfile'),
  },
  {
    icon: <MailOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.profile.bindEmail'),
  },
];
export const UserSettingsTabDataSource: () => ITabWrapperDataSource[] = () => [
  {
    icon: <UserOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.settings.accountSetting'),
  },
  {
    icon: <LockOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.settings.permissionSetting'),
  },
  {
    icon: <FormOutlined />,
    onClick: () => undefined,
    text: i18n.t('layout.header.userNavigation.profile.modifyPassword'),
  },
];

export default UserNavigation;
