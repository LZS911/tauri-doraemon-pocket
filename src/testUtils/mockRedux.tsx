import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import layout from '../store/layout';
import config from '../store/config';
import { Dictionary } from '../typing/common.type';
import { Provider } from 'react-redux';
import React, { ReactNode } from 'react';
const reducers = combineReducers({
  layout,
  config,
});
export const storeFactory: any = (initStore: Dictionary = {}) => {
  return configureStore({
    reducer: reducers,
    preloadedState: initStore,
  });
};

export const CustomProvider: React.FC<{
  initStore: Dictionary;
  children: ReactNode;
}> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
