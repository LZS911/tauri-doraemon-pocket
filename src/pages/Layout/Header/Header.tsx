import { MenuFoldOutlined, MenuUnfoldOutlined } from '@ant-design/icons';
import ThemeBase from '../../../components/ThemeBase';
import useLayoutRedux from '../useLayoutRedux';
import useResizeObserver from 'use-resize-observer';
import { useBoolean } from 'ahooks';
import useThrottle from '../../../hooks/useThrottle';
import ToggleLanguage from './ToggleLanguage';
import Customization from './Customization';
import Notification from './Notification';
import UserNavigation from './UserNavigation';

const Header: React.FC = () => {
  const { isExpandSider, setIsExpandSider } = useLayoutRedux();
  const headerWidthObserver = useThrottle(
    ({ width }: { width?: number }) => {
      if ((width ?? 0) < 460) {
        hideUserInfo();
      } else {
        showUserInfo();
      }
    },
    [],
    100
  );
  const { ref } = useResizeObserver<HTMLDivElement>({
    onResize: headerWidthObserver,
  });
  const [
    userInfoVisibility,
    { setFalse: hideUserInfo, setTrue: showUserInfo },
  ] = useBoolean(true);

  const handleClickMenuIcon = (isExpand: boolean) => {
    setIsExpandSider(!isExpand);
  };

  return (
    <header
      ref={ref}
      className="sticky right-0 top-0 z-[1000] border-b border-stone-100 dark:border-stone-800"
    >
      <ThemeBase.Paper
        className={`flex w-full items-center justify-between py-2 pl-2 pr-6`}
      >
        <ThemeBase.Icon
          data-testid="menu-icon"
          icon={
            isExpandSider ? (
              <MenuFoldOutlined className="text-[1rem]" />
            ) : (
              <MenuUnfoldOutlined className="text-[1rem]" />
            )
          }
          onClick={() => handleClickMenuIcon(isExpandSider)}
          className="bg-slate-100 dark:bg-black"
        />

        <div className="flex items-center">
          <ToggleLanguage />

          {/* <Notification /> */}

          <Customization />

          {userInfoVisibility && <UserNavigation />}
        </div>
      </ThemeBase.Paper>
    </header>
  );
};

export default Header;
