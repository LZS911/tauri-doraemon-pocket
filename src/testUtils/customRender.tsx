/* eslint-disable @typescript-eslint/no-explicit-any */
import { render, renderHook } from '@testing-library/react';
import { Provider } from 'react-redux';
import {
  BrowserRouter,
  MemoryRouter,
  MemoryRouterProps,
  Router,
  RouterProps,
} from 'react-router-dom';
import { Dictionary } from '../typing/common.type';
import { storeFactory } from './mockRedux';
type RenderParams = Parameters<typeof render>;

export const renderWithRedux: (arg: any) => ReturnType<typeof render> = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};
export const renderWithRouter: any = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter: any = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithServerRouterAndRedux: any = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      {' '}
      <Provider store={storeFactory()}>{ui}</Provider>,
    </Router>,
    option
  );
};

export const renderWithMemoryRouter: any = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};
