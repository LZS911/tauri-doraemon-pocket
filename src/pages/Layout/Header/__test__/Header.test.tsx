import { fireEvent, screen, waitFor } from '@testing-library/react';
import { useRef } from 'react';
import { useDispatch } from 'react-redux';
import useResizeObserver from 'use-resize-observer';
import { renderWithRedux } from '../../../../testUtils/customRender';
import Header from '../index';

jest.mock('use-resize-observer', () => {
  return {
    default: jest.fn(),
  };
});

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
  };
});

describe('test Header', () => {
  const useResizeObserverMock: jest.Mock = useResizeObserver as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;
  const dispatchMock = jest.fn();

  beforeEach(() => {
    jest.useFakeTimers();
    useDispatchMock.mockImplementation(() => dispatchMock);
    useResizeObserverMock.mockImplementation(() => {
      return {
        ref: useRef(),
      };
    });
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
    jest.clearAllTimers();
  });
  test('should match snapshot', async () => {
    const { container } = renderWithRedux(<Header />);
    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });
    expect(container).toMatchSnapshot();
  });

  test('should dispatch setIsExpandSider action when clicking menu icon', async () => {
    renderWithRedux(<Header />);

    await waitFor(() => {
      jest.advanceTimersByTime(0);
    });

    expect(dispatchMock).toBeCalledTimes(0);
    fireEvent.click(screen.getByTestId('menu-icon'));
    expect(dispatchMock).toBeCalledTimes(1);
    expect(dispatchMock).toBeCalledWith({
      payload: true,
      type: 'layout/setIsExpandSider',
    });
  });
});
