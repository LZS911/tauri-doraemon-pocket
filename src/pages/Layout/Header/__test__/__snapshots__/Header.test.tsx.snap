// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Header should match snapshot 1`] = `
<div>
  <header
    class="sticky top-0 right-0 z-[1000] border-b border-stone-100 dark:border-stone-800"
  >
    <div
      class="bg-white dark:bg-darkMode dark:text-white flex w-full items-center justify-between py-2 pr-6 pl-2"
    >
      <div
        class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
        data-testid="menu-icon"
      >
        <span
          aria-label="menu-unfold"
          class="anticon anticon-menu-unfold text-[1rem]"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="menu-unfold"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
            />
          </svg>
        </span>
      </div>
      <div
        class="flex items-center"
      >
        <div>
          <div
            class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-2 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
            data-testid="open-language-options"
          >
            <span
              aria-label="translation"
              class="anticon anticon-translation text-[1rem]"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="translation"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <defs>
                  <style />
                </defs>
                <path
                  d="M140 188h584v164h76V144c0-17.7-14.3-32-32-32H96c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h544v-76H140V188z"
                />
                <path
                  d="M414.3 256h-60.6c-3.4 0-6.4 2.2-7.6 5.4L219 629.4c-.3.8-.4 1.7-.4 2.6 0 4.4 3.6 8 8 8h55.1c3.4 0 6.4-2.2 7.6-5.4L322 540h196.2L422 261.4a8.42 8.42 0 00-7.7-5.4zm12.4 228h-85.5L384 360.2 426.7 484zM936 528H800v-93c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v93H592c-13.3 0-24 10.7-24 24v176c0 13.3 10.7 24 24 24h136v152c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V752h136c13.3 0 24-10.7 24-24V552c0-13.3-10.7-24-24-24zM728 680h-88v-80h88v80zm160 0h-88v-80h88v80z"
                />
              </svg>
            </span>
          </div>
        </div>
        <div
          class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-2 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
          data-testid="popover-switch-icon"
        >
          <span
            class="ant-badge ant-badge-status leading-[0.875rem]"
          >
            <span
              class="dark:text-white"
            >
              <span
                aria-label="bell"
                class="anticon anticon-bell text-[1rem]"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="bell"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </div>
        <div
          class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show mr-7 bg-slate-100 dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
          data-testid="draw-switch-icon"
        >
          <span
            aria-label="setting"
            class="anticon anticon-setting text-[1rem]"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="setting"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 009.3-35.2l-.9-2.6a443.74 443.74 0 00-79.7-137.9l-1.8-2.1a32.12 32.12 0 00-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 00-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 00-25.8 25.7l-15.8 85.4a351.86 351.86 0 00-99 57.4l-81.9-29.1a32 32 0 00-35.1 9.5l-1.8 2.1a446.02 446.02 0 00-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 00-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0035.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0025.8 25.7l2.7.5a449.4 449.4 0 00159 0l2.7-.5a32.05 32.05 0 0025.8-25.7l15.7-85a350 350 0 0099.7-57.6l81.3 28.9a32 32 0 0035.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 01-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 01-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 01624 502c0 29.9-11.7 58-32.8 79.2z"
              />
            </svg>
          </span>
        </div>
        <div
          class="hover:bg-slate-200 dark:hover:bg-slate-400 after:water-wave-hide active:after:water-wave-show bg-slate-100 px-2 py-1 text-base dark:bg-black transparent relative flex cursor-pointer items-center justify-center rounded-md bg-transparent p-2 transition-colors dark:text-white"
          data-testid="open-popover-icon"
        >
          <img
            class="h-6 w-6 rounded-[50%] bg-primary"
            src="https://joeschmoe.io/api/v1/random"
          />
          <span
            class="ml-3 font-bold"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
